proxy/gateway/native/devdoc/proxy_gateway_requirements.md

```diff
@@ -0,0 +1,184 @@
+Module Requirements
+===================
+
+## Overview
+
+This is the documentation for a remote module that communicates with other
+modules through the Azure IoT Gateway. Remote modules need to implement the
+well defined Module Interface. Each remote module will additionally
+need to attach and detach from the gateway and will either need to manage
+the worker thread manually or use the convenience methods supplied to start
+and halt the worker thread.
+
+## References
+
+[Azure IoT Gateway Module Interface](../../../../core/devdoc/module.md)
+
+![High Level Design Diagrams](./media/proxy_gateway_hld.PNG)
+
+## Exposed Types
+
+```c
+typedef struct REMOTE_MODULE_TAG * REMOTE_MODULE_HANDLE;
+```
+
+## Exposed API
```

Comment:
We usually include the function definitions up here as well, under the idea that more information here is better.